src/services/orderServices/updateOrderService.ts

```diff
@@ -0,0 +1,127 @@
+import { Request, Response } from 'express';
+import { getManager, EntityManager, Repository } from 'typeorm';
+import { Order } from '../../entities/Order';
+import { Product } from '../../entities/Product';
+import { User } from '../../entities/User';
+import { OrderItem } from '../../entities/OrderItem';
+import { Transaction } from '../../entities/transaction';
+import { responseError, sendErrorResponse, sendSuccessResponse } from '../../utils/response.utils';
+import sendMail from '../../utils/sendOrderMail';
+
+export const updateOrderService = async (req: Request, res: Response) => {
+    const { orderId } = req.params;
+    const { orderStatus } = req.body;
+
+    try {
+        await getManager().transaction(async (transactionalEntityManager: EntityManager) => {
+            const orderRepository: Repository<Order> = transactionalEntityManager.getRepository(Order);
+            const productRepository: Repository<Product> = transactionalEntityManager.getRepository(Product);
+            const userRepository: Repository<User> = transactionalEntityManager.getRepository(User);
+            const orderItemRepository: Repository<OrderItem> = transactionalEntityManager.getRepository(OrderItem);
+            const transactionRepository: Repository<Transaction> = transactionalEntityManager.getRepository(Transaction);
+
+            const buyerId = req.user?.id;
+            if (!buyerId) {
+                throw new Error('Unauthorized');
+            }
+
+            // Fetch order and related entities
+            const order: Order | null = await orderRepository.findOne({
+                where: { id: orderId, buyer: { id: buyerId } },
+                relations: ['orderItems', 'orderItems.product', 'buyer'],
+            });
+
+            if (!order) {
+                return sendErrorResponse(res, 404, "Order not found");
+            }
+            // Check if order can be updated
+            if (isOrderFinalStatus(order.orderStatus)) {
+                return sendErrorResponse(res, 401, `Order cannot be updated once it is ${order.orderStatus}`);
+            }
+
+            // Handle order status transitions
+            if (orderStatus !== undefined && order.orderStatus !== orderStatus) {
```

Comment:
@elijahladdie , @UwicyezaG , Before updating the order status, could you please check if the order has been marked as "received" first? It doesn't make sense for a buyer to update an order to "canceled" after they have already confirmed that they received it.